**Gregg Maloy**   
**Data 602**   
**Assignment_6**

In this assignment, you will practice loading data into your python environment. It is recommended that you use either google colab or juypter notebook, however, you may use any environment you prefer as long as you are using the pandas library.   
From the following link, choose two files to answer the questions below.   
https://github.com/data602sps/datasetspractice      

In addition, you will also search the internet to find two datasets to answer the questions below.   

Websites to find data:   
● https://opendata.cityofnewyork.us/   
● https://datasetsearch.research.google.com/   
● https://www.kaggle.com/datasets   
● http://archive.ics.uci.edu/ml/index.php   
● https://data.gov/   

Q1. Load your dataset into python using the pandas library and save data into a   
dataframe named dfi (where i is one of your datasets, for a total of 4).   
Q2. Preview your data by calling your dataframe’s name. How many columns and rows   
do you see?   
Q3. Examine the shape of your data using the .shape command and the data types of   
your columns using .dtypes.   
Q4. Use .describe() on your data. What do you notice about your data? What does this   
command return?   
Q5. Use the .head() and .tail() command - what does this do?

**WEATHER DATA**

Q1. Load your dataset into python using the pandas library and save data into a
dataframe named dfi (where i is one of your datasets, for a total of 4).

In [120]:
import pandas as pd
url='https://raw.githubusercontent.com/data602sps/datasetspractice/main/weather.csv'
dfi=pd.read_csv(url)

Q2. Preview your data by calling your dataframe’s name. How many columns and rows
do you see?

In [121]:
dfi

Unnamed: 0 origin  year  month   day  hour   temp   dewp  humid  \
0              1    EWR  2013    1.0   1.0   0.0  37.04  21.92  53.97   
1              2    EWR  2013    1.0   1.0   1.0  37.04  21.92  53.97   
2              3    EWR  2013    1.0   1.0   2.0  37.94  21.92  52.09   
3              4    EWR  2013    1.0   1.0   3.0  37.94  23.00  54.51   
4              5    EWR  2013    1.0   1.0   4.0  37.94  24.08  57.04   
...          ...    ...   ...    ...   ...   ...    ...    ...    ...   
8714        8715    JFK  2013    9.0   2.0  20.0  75.20  73.40  94.14   
8715        8716    JFK  2013   10.0  23.0  10.0  48.92  39.02  68.51   
8716        8717    JFK  2013   10.0  23.0  11.0  48.92  39.02  68.51   
8717        8718    JFK  2013   12.0  17.0   5.0  26.96  10.94  50.34   
8718        8719    LGA  2013    8.0  22.0  22.0  75.92  66.92  73.68   

      wind_dir  wind_speed  wind_gust  precip  pressure  visib  
0        230.0    10.35702  11.918651     0.0    1013.9   10.0  
1        230.0    13.80936  15.891535     0.0    1013.0   10.0  
2        230.0    12.65858  14.567241     0.0    1012.6   10.0  
3        230.0    13.80936  15.891535     0.0    1012.7   10.0  
4        240.0    14.96014  17.215830     0.0    1012.8   10.0  
...        ...         ...        ...     ...       ...    ...  
8714     200.0     4.60312   5.297178     0.0       NaN    4.0  
8715      60.0     4.60312   5.297178     0.0    1008.1   10.0  
8716      40.0     4.60312   5.297178     0.0    1008.5   10.0  
8717      40.0     4.60312   5.297178     0.0    1023.9   10.0  
8718     210.0     8.05546   9.270062     0.0    1011.9   10.0  

[8719 rows x 15 columns]

Q3. Examine the shape of your data using the .shape command and the data types of
your columns using .dtypes.

In [122]:
# shape
data_shape = dfi.shape
print("Shape of dfi (# of rows,# of columns): ",data_shape)

# type
data_types = dfi.dtypes
print("Data types of dfi: ",data_types)


Shape of dfi (# of rows,# of columns):  (8719, 15)
Data types of dfi:  Unnamed: 0      int64
origin         object
year            int64
month         float64
day           float64
hour          float64
temp          float64
dewp          float64
humid         float64
wind_dir      float64
wind_speed    float64
wind_gust     float64
precip        float64
pressure      float64
visib         float64
dtype: object


In [123]:
print(dfi.shape)

(8719, 15)


Q4. Use .describe() on your data. What do you notice about your data? What does this
command return?

In [124]:
dfi.describe()

Unnamed: 0    year        month          day         hour  \
count  8719.000000  8719.0  8718.000000  8718.000000  8718.000000   
mean   4360.000000  2013.0     6.506538    15.679628    11.515715   
std    2517.102832     0.0     3.439961     8.768250     6.916846   
min       1.000000  2013.0     1.000000     1.000000     0.000000   
25%    2180.500000  2013.0     4.000000     8.000000     6.000000   
50%    4360.000000  2013.0     7.000000    16.000000    12.000000   
75%    6539.500000  2013.0     9.000000    23.000000    18.000000   
max    8719.000000  2013.0    12.000000    31.000000    23.000000   

              temp         dewp        humid     wind_dir   wind_speed  \
count  8718.000000  8718.000000  8718.000000  8486.000000  8718.000000   
mean     55.484942    41.794184    62.863333   193.194674     9.335732   
std      18.345817    19.619776    19.572699   110.541333    12.328261   
min      10.940000    -9.040000    13.950000     0.000000     0.000000   
25%      39.920000    26.060000    47.602500   110.000000     5.753900   
50%      55.940000    42.800000    62.330000   220.000000     9.206240   
75%      71.060000    57.920000    78.790000   280.000000    12.658580   
max     100.040000    75.920000   100.000000   360.000000  1048.360580   

         wind_gust       precip     pressure        visib  
count  8718.000000  8719.000000  7780.000000  8719.000000  
mean     10.743374     0.003035  1017.826465     9.224522  
std      14.187117     0.022686     7.396722     2.094044  
min       0.000000     0.000000   983.900000     0.120000  
25%       6.621473     0.000000  1012.800000    10.000000  
50%      10.594357     0.000000  1017.500000    10.000000  
75%      14.567241     0.000000  1022.900000    10.000000  
max    1206.432388     1.180000  1041.900000    10.000000

What does this command return?   
The '.describe' provides measures of central tendency, spread, and distribution of dfi.

What do you notice about your data?   
There are 8719 rows with variables including year, month, day and hour.
All measurements are recording in 2013.
Looks like the measurements are pretty evenly distributed as 50% of medians fall where would be expected in terms of month, day and hour 50% month=July, 50% day=16th, 50% hour=12).
A quick glance at max and min denotes a good spread of measurements (ie temp min=10.9, temp max=100, 50% median=55).
The outlyiers that become apparent are: 
-max wind_speed (max=1048) while 50% median=9.2, and 
-max wind gust (max=1206) while 50% median=10.5.
Also of note precipitation looks to be underrepresented possibly.

Q5. Use the .head() and .tail() command - what does this do?

In [125]:
dfi.head()

Unnamed: 0 origin  year  month  day  hour   temp   dewp  humid  wind_dir  \
0           1    EWR  2013    1.0  1.0   0.0  37.04  21.92  53.97     230.0   
1           2    EWR  2013    1.0  1.0   1.0  37.04  21.92  53.97     230.0   
2           3    EWR  2013    1.0  1.0   2.0  37.94  21.92  52.09     230.0   
3           4    EWR  2013    1.0  1.0   3.0  37.94  23.00  54.51     230.0   
4           5    EWR  2013    1.0  1.0   4.0  37.94  24.08  57.04     240.0   

   wind_speed  wind_gust  precip  pressure  visib  
0    10.35702  11.918651     0.0    1013.9   10.0  
1    13.80936  15.891535     0.0    1013.0   10.0  
2    12.65858  14.567241     0.0    1012.6   10.0  
3    13.80936  15.891535     0.0    1012.7   10.0  
4    14.96014  17.215830     0.0    1012.8   10.0

In [126]:
dfi.tail()

Unnamed: 0 origin  year  month   day  hour   temp   dewp  humid  \
8714        8715    JFK  2013    9.0   2.0  20.0  75.20  73.40  94.14   
8715        8716    JFK  2013   10.0  23.0  10.0  48.92  39.02  68.51   
8716        8717    JFK  2013   10.0  23.0  11.0  48.92  39.02  68.51   
8717        8718    JFK  2013   12.0  17.0   5.0  26.96  10.94  50.34   
8718        8719    LGA  2013    8.0  22.0  22.0  75.92  66.92  73.68   

      wind_dir  wind_speed  wind_gust  precip  pressure  visib  
8714     200.0     4.60312   5.297178     0.0       NaN    4.0  
8715      60.0     4.60312   5.297178     0.0    1008.1   10.0  
8716      40.0     4.60312   5.297178     0.0    1008.5   10.0  
8717      40.0     4.60312   5.297178     0.0    1023.9   10.0  
8718     210.0     8.05546   9.270062     0.0    1011.9   10.0

The '.head()' and '.tail()' commands display the first five rows and last 5 rows of the df respectively.

**Salary Data**

Q1. Load your dataset into python using the pandas library and save data into a dataframe named dfi (where i is one of your datasets, for a total of 4).

In [127]:
import pandas as pd
url='https://raw.githubusercontent.com/data602sps/datasetspractice/main/salarydata.csv'
dfii=pd.read_csv(url)

Q2. Preview your data by calling your dataframe’s name. How many columns and rows
do you see?

In [128]:
dfii

Sate  2018 Mean Annual Wage  Unnamed: 2
0               Alabama                 90,620         NaN
1                 Alaska                99,180         NaN
2                Arizona                88,800         NaN
3               Arkansas                81,430         NaN
4             California                97,110         NaN
5               Colorado                82,560         NaN
6            Connecticut                96,010         NaN
7               Delaware                93,880         NaN
8   District of Columbia                89,880         NaN
9                Florida                87,410         NaN
10               Georgia                86,320         NaN
11                Hawaii                90,540         NaN
12                 Idaho                77,700         NaN
13              IIIinois                90,690         NaN
14               lndiana                83,680         NaN
15                  lowa                82,960         NaN
16                Kansas                85,250         NaN
17              Kentucky                84,630         NaN
18             Louisiana                89,860         NaN
19                 Maine                79,910         NaN
20              Maryland                85,170         NaN
21         Massachusetts                91,750         NaN
22              Michigan                91,160         NaN
23             Minnesota                83,750         NaN
24           Mississippi                89,720         NaN
25              Missouri                81,330         NaN
26               Montana                79,050         NaN
27              Nedraska                80,130         NaN
28                Nevada               107,920         NaN
29         New Hampshire                82,880         NaN
30            New Jersey                97,770         NaN
31           New  Mexico                97,210         NaN
32              New York                87,470         NaN
33        North Carolina                87,560         NaN
34          North Dakota                78,120         NaN
35                  Ohio                86,690         NaN
36              Okahoma                 84,860         NaN
37                Oregon                85,890         NaN
38          Pennsylvania                87,050         NaN
39          Rhode Island                83,850         NaN
40        South Carolina                85,450         NaN
41          South Dakota                92,940         NaN
42             Tennessee                85,940         NaN
43                 Taxas                92,940         NaN
44                  Utah                85,940         NaN
45               Vermont                75,010         NaN
46              Virginia                91,700         NaN
47            Washington                85,930         NaN
48         West Virginia                89,420         NaN
49             Wisconsin                85,200         NaN
50               Wyomimg                87,510         NaN

50 rows and 3 columns.
One column consists entirely of NaN data, so really only two columns

Q3. Examine the shape of your data using the .shape command and the data types of
your columns using .dtypes.

In [129]:
# shape
data_shape = dfii.shape
print("Shape of dfi (# of rows,# of columns): ",data_shape)

# type
data_types = dfii.dtypes
print("Data types of dfi: ",data_types)

Shape of dfi (# of rows,# of columns):  (51, 3)
Data types of dfi:  Sate                      object
2018 Mean Annual Wage     object
Unnamed: 2               float64
dtype: object


Q4. Use .describe() on your data. What do you notice about your data? What does this
command return?

Some minor wrangling was necessarly to use the '.describe' command.  
Initially, as noted below, the '.describe()' command did not return any results because the third column of the dataset was all 'NaN'.  
The third column was 'dropped'

In [130]:
dfii.describe()

Unnamed: 2
count         0.0
mean          NaN
std           NaN
min           NaN
25%           NaN
50%           NaN
75%           NaN
max           NaN

In [131]:
#dropping the third column of NaN 
dfii.drop("Unnamed: 2", axis=1, inplace=True)

In [132]:
dfii.describe()

Sate  2018 Mean Annual Wage
count         51                    51
unique        51                    49
top     Alabama                 85,940
freq           1                     2

The '.describe' command did return more information, because  '2018 Mean Annual Wage' variable is a 'object', central tendancy measures are not returned. Below the '2018 Mean Annual Wage' is chnaged to a 'float'

In [133]:
dfii["2018 Mean Annual Wage"].isna()
dfii["2018 Mean Annual Wage"].info()
dfii["2018 Mean Annual Wage"] = dfii["2018 Mean Annual Wage"].str.replace('[\$,]', '', regex=True).astype(float)
dfii["2018 Mean Annual Wage"].info()

<class 'pandas.core.series.Series'>
RangeIndex: 51 entries, 0 to 50
Series name: 2018 Mean Annual Wage
Non-Null Count  Dtype 
--------------  ----- 
51 non-null     object
dtypes: object(1)
memory usage: 536.0+ bytes
<class 'pandas.core.series.Series'>
RangeIndex: 51 entries, 0 to 50
Series name: 2018 Mean Annual Wage
Non-Null Count  Dtype  
--------------  -----  
51 non-null     float64
dtypes: float64(1)
memory usage: 536.0 bytes


In [134]:
dfii.describe()

2018 Mean Annual Wage
count              51.000000
mean            87563.333333
std              6117.818456
min             75010.000000
25%             83800.000000
50%             86690.000000
75%             90655.000000
max            107920.000000

Convcerting to 'float' worked.  The final dataset has 51 rows and 2 columns. Mean annual wage = $87,563.33 with a minimum value of $75,010 and a max value of $107,920. 

Q5. Use the .head() and .tail() command - what does this do?

In [135]:
#returns top 3 rows of dfii
dfii.head(3)

Sate   2018 Mean Annual Wage
0  Alabama                 90620.0
1    Alaska                99180.0
2   Arizona                88800.0

In [136]:
#returns bottom 3 rows of dfii
dfii.tail(3)

Sate   2018 Mean Annual Wage
48  West Virginia                89420.0
49      Wisconsin                85200.0
50        Wyomimg                87510.0

**Social Network Analysis Dataset from Kaggle**

Q1. Load your dataset into python using the pandas library and save data into a
dataframe named dfi (where i is one of your datasets, for a total of 4).

In [137]:
import pandas as pd
url='https://raw.githubusercontent.com/greggmaloy/Data620/main/DATA_620_Assignment_1.csv'
dfiii=pd.read_csv(url)

Q2. Preview your data by calling your dataframe’s name. How many columns and rows
do you see?

In [138]:
dfiii

Organisation                 member
0         ethereum          Oleh Aldekein
1         ethereum                   None
2         ethereum     Frank Szendzielarz
3         ethereum      Kolby Moroz Liebl
4         ethereum  Marius van der Wijden
...            ...                    ...
10593    microsoft           Zachary Danz
10594    microsoft                    zsy
10595    microsoft          Zuhair Parvez
10596    microsoft                   None
10597    microsoft              Yiwen Zhu

[10598 rows x 2 columns]

Q3. Examine the shape of your data using the .shape command and the data types of
your columns using .dtypes.

In [139]:
# shape
data_shape = dfiii.shape
print("Shape of dfi (# of rows,# of columns): ",data_shape)

# type
data_types = dfiii.dtypes
print("Data types of dfi: ",data_types)

Shape of dfi (# of rows,# of columns):  (10598, 2)
Data types of dfi:  Organisation    object
member          object
dtype: object


Q4. Use .describe() on your data. What do you notice about your data? What does this
command return?

In [140]:
dfiii.describe()

Organisation member
count         10598  10598
unique           41   8879
top       microsoft   None
freq           4362    946

In [141]:
#below i perform some minor wrangling to provide more information about the data
organization_member_counts = dfiii.groupby('Organisation')['member'].count()
organization_member_counts = organization_member_counts.sort_values(ascending=False)
print(organization_member_counts)

Organisation
microsoft       4362
Azure           1931
apache          1123
google           877
nodejs           370
aws              180
elastic          173
reactjs          147
python           119
facebook         110
ethereum          84
freeCodeCamp      74
flutter           71
babel             70
rails             65
jquery            65
grafana           64
golang            63
django            62
vuejs             55
mongodb           55
pytorch           51
huggingface       49
ansible           47
atom              47
firebase          41
serverless        38
bitcoin           30
prometheus        26
airbnb            26
deepmind          21
docker            18
graphql           17
NVIDIA            12
laravel           11
plotly            11
opencv            10
nuxt               7
reduxjs            6
redis              5
udacity            5
Name: member, dtype: int64


The dataset consists of github 'Organisations' which are developer groups on github.   
Each row represents one member of an organisation.
There are 10,598 rows(members) and 41 'organisations'.
Microsoft is the organisation with the most members, 4362.

In [142]:
Q5. Use the .head() and .tail() command - what does this do?

Object `do` not found.


In [143]:
dfiii.head(5)

Organisation                 member
0     ethereum          Oleh Aldekein
1     ethereum                   None
2     ethereum     Frank Szendzielarz
3     ethereum      Kolby Moroz Liebl
4     ethereum  Marius van der Wijden

In [144]:
dfiii.tail(5)

Organisation         member
10593    microsoft   Zachary Danz
10594    microsoft            zsy
10595    microsoft  Zuhair Parvez
10596    microsoft           None
10597    microsoft      Yiwen Zhu

**Bed Rug Reporting - NYC Open Data**   
I travel a lot and bed bugs have been a recent media topic.  
I decided to query NYC Open Data to find cases of bedbug infestation in buildings in the zip code I live.
The zip code is 10463, which is in Riverdale, Bronx.

Q1. Load your dataset into python using the pandas library and save data into a
dataframe named dfi (where i is one of your datasets, for a total of 4).

In [145]:
import pandas as pd
url='https://raw.githubusercontent.com/greggmaloy/Data_602/main/assignment_6_Bedbug_Reporting.csv'
dfiv=pd.read_csv(url)

Q2. Preview your data by calling your dataframe’s name. How many columns and rows
do you see?

In [146]:
dfiv

Building ID  Registration ID Borough House Number  \
0          806228           221419   BRONX         3130   
1           44824           206896   BRONX         3044   
2           44826           202473   BRONX         3060   
3           44826           202473   BRONX         3060   
4           44824           206896   BRONX         3044   
...           ...              ...     ...          ...   
1657       968093           227740   BRONX          164   
1658       876582           223120   BRONX          201   
1659      1007815           228614   BRONX          157   
1660       968094           227738   BRONX          166   
1661       876582           223120   BRONX          201   

                Street Name  Postcode  # of Dwelling Units  \
0           ALBANY CRESCENT     10463                   58   
1           ALBANY CRESCENT     10463                   36   
2           ALBANY CRESCENT     10463                   20   
3           ALBANY CRESCENT     10463                   20   
4           ALBANY CRESCENT     10463                   36   
...                     ...       ...                  ...   
1657  WEST KINGSBRIDGE ROAD     10463                   10   
1658  WEST KINGSBRIDGE ROAD     10463                   55   
1659  WEST KINGSBRIDGE ROAD     10463                   20   
1660  WEST KINGSBRIDGE ROAD     10463                   10   
1661  WEST KINGSBRIDGE ROAD     10463                   54   

      Infested Dwelling Unit Count  Eradicated Unit Count  \
0                                0                      0   
1                                0                      0   
2                                0                      0   
3                                1                      1   
4                                0                      0   
...                            ...                    ...   
1657                             0                      0   
1658                             0                      0   
1659                             0                      0   
1660                             0                      0   
1661                             2                      2   

      Re-infested  Dwelling Unit Count  ... Filing Period Start Date  \
0                                    0  ...                11/1/2017   
1                                    0  ...                11/1/2020   
2                                    0  ...                11/1/2020   
3                                    0  ...                11/1/2017   
4                                    0  ...                11/1/2021   
...                                ...  ...                      ...   
1657                                 0  ...                11/1/2020   
1658                                 0  ...                11/1/2020   
1659                                 0  ...                11/1/2021   
1660                                 0  ...                11/1/2021   
1661                                 0  ...                11/1/2019   

     Filling Period End Date   Latitude  Longitude  Community Board  \
0                 11/30/2018  40.878970 -73.903116                8   
1                 10/31/2021  40.877233 -73.903650                8   
2                 10/31/2021  40.877521 -73.903765                8   
3                 11/30/2018  40.877521 -73.903765                8   
4                 10/31/2022  40.877233 -73.903650                8   
...                      ...        ...        ...              ...   
1657              10/31/2021  40.870828 -73.904491                7   
1658              10/31/2021  40.871552 -73.904446                8   
1659              10/31/2022  40.870130 -73.904192                8   
1660              10/31/2022  40.870877 -73.904487                7   
1661              10/31/2020  40.871552 -73.904446                8   

      Council District  2010 Census Tract        BIN           BBL  \
0                   14         

In [147]:
Q3. Examine the shape of your data using the .shape command and the data types of
your columns using .dtypes.

SyntaxError: invalid syntax (898685247.py, line 1)

In [148]:
# shape
data_shape = dfiv.shape
print("Shape of dfi (# of rows,# of columns): ",data_shape)

# type
data_types = dfiv.dtypes
print("Data types of dfi: ",data_types)

Shape of dfi (# of rows,# of columns):  (1662, 21)
Data types of dfi:  Building ID                           int64
Registration ID                       int64
Borough                              object
House Number                         object
Street Name                          object
Postcode                              int64
# of Dwelling Units                   int64
Infested Dwelling Unit Count          int64
Eradicated Unit Count                 int64
Re-infested  Dwelling Unit Count      int64
Filing Date                          object
Filing Period Start Date             object
Filling Period End Date              object
Latitude                            float64
Longitude                           float64
Community Board                       int64
Council District                      int64
2010 Census Tract                     int64
BIN                                 float64
BBL                                 float64
NTA                                  object
dtype

Q4. Use .describe() on your data. What do you notice about your data? What does this
command return?

In [149]:
dfiv.describe()

Building ID  Registration ID  Postcode  # of Dwelling Units  \
count  1.662000e+03      1662.000000    1662.0          1662.000000   
mean   2.322170e+05    224765.721420   10463.0            64.145608   
std    3.094813e+05     70043.753709       0.0            68.091683   
min    4.482200e+04    200300.000000   10463.0             1.000000   
25%    8.421100e+04    207421.000000   10463.0            17.000000   
50%    8.982900e+04    217307.000000   10463.0            55.000000   
75%    1.214780e+05    221744.000000   10463.0            81.000000   
max    1.007815e+06    714898.000000   10463.0           589.000000   

       Infested Dwelling Unit Count  Eradicated Unit Count  \
count                   1662.000000            1662.000000   
mean                       0.285199               0.506619   
std                        1.284562               4.572082   
min                        0.000000               0.000000   
25%                        0.000000               0.000000   
50%                        0.000000               0.000000   
75%                        0.000000               0.000000   
max                       32.000000             120.000000   

       Re-infested  Dwelling Unit Count     Latitude    Longitude  \
count                       1662.000000  1662.000000  1662.000000   
mean                           0.037304    40.881690   -73.906088   
std                            0.396923     0.004933     0.006810   
min                            0.000000    40.869871   -73.922381   
25%                            0.000000    40.878548   -73.910439   
50%                            0.000000    40.882527   -73.905365   
75%                            0.000000    40.885190   -73.902545   
max                           12.000000    40.889914   -73.889344   

       Community Board  Council District  2010 Census Tract           BIN  \
count      1662.000000       1662.000000        1662.000000  1.659000e+03   
mean          7.967509         11.707581        4481.430806  2.065720e+06   
std           0.177353          1.273989       10315.763643  3.660800e+04   
min           7.000000         11.000000         269.000000  2.000000e+06   
25%           8.000000         11.000000         279.000000  2.015996e+06   
50%           8.000000         11.000000         287.000000  2.083558e+06   
75%           8.000000         11.000000         297.000000  2.085778e+06   
max           8.000000         14.000000       30701.000000  2.130523e+06   

                BBL  
count  1.659000e+03  
mean   2.047539e+09  
std    1.239097e+07  
min    2.032370e+09  
25%    2.032600e+09  
50%    2.057230e+09  
75%    2.057791e+09  
max    2.059201e+09

The describe command for this particular dataset is not particularlly useful because most of the data is address data and therefore not useful for central tendencies.  The exceptions maybe for '# of Dwelling Units' and 'Infested Dwelling Unit Count'.  '# of Dwelling Units' may be useful because it provides potential information about poulation density, while 'Infested Dwelling Unit Count' provides information about the number of units infested in a building.

In [150]:
#below i perform some minor wrangling to provide more information about the data
Buildings_infestation_counts = dfiv.groupby('Building ID')['Infested Dwelling Unit Count'].count()
Buildings_infestation_counts = Buildings_infestation_counts.sort_values(ascending=False)
df_counts = pd.DataFrame(Buildings_infestation_counts)
df_counts.reset_index(inplace=True)
df_counts.columns = ['Building ID', 'Infested Dwelling Unit Count']
print(df_counts)


     Building ID  Infested Dwelling Unit Count
0          46499                             9
1          89781                             8
2         875578                             8
3          88072                             7
4          84284                             7
..           ...                           ...
448       892725                             1
449        82197                             1
450        82196                             1
451       885298                             1
452      1007815                             1

[453 rows x 2 columns]


Q5. Use the .head() and .tail() command - what does this do?

In [151]:
dfiv.head()

Building ID  Registration ID Borough House Number      Street Name  \
0       806228           221419   BRONX         3130  ALBANY CRESCENT   
1        44824           206896   BRONX         3044  ALBANY CRESCENT   
2        44826           202473   BRONX         3060  ALBANY CRESCENT   
3        44826           202473   BRONX         3060  ALBANY CRESCENT   
4        44824           206896   BRONX         3044  ALBANY CRESCENT   

   Postcode  # of Dwelling Units  Infested Dwelling Unit Count  \
0     10463                   58                             0   
1     10463                   36                             0   
2     10463                   20                             0   
3     10463                   20                             1   
4     10463                   36                             0   

   Eradicated Unit Count  Re-infested  Dwelling Unit Count  ...  \
0                      0                                 0  ...   
1                      0                                 0  ...   
2                      0                                 0  ...   
3                      1                                 0  ...   
4                      0                                 0  ...   

  Filing Period Start Date Filling Period End Date   Latitude  Longitude  \
0                11/1/2017              11/30/2018  40.878970 -73.903116   
1                11/1/2020              10/31/2021  40.877233 -73.903650   
2                11/1/2020              10/31/2021  40.877521 -73.903765   
3                11/1/2017              11/30/2018  40.877521 -73.903765   
4                11/1/2021              10/31/2022  40.877233 -73.903650   

   Community Board  Council District  2010 Census Tract        BIN  \
0                8                14                277  2016147.0   
1                8                14                277  2016127.0   
2                8                14                277  2016129.0   
3                8                14                277  2016129.0   
4                8                14                277  2016127.0   

            BBL                                        NTA  
0  2.032670e+09  Kingsbridge Heights-Van Cortlandt Village  
1  2.032660e+09  Kingsbridge Heights-Van Cortlandt Village  
2  2.032660e+09  Kingsbridge Heights-Van Cortlandt Village  
3  2.032660e+09  Kingsbridge Heights-Van Cortlandt Village  
4  2.032660e+09  Kingsbridge Heights-Van Cortlandt Village  

[5 rows x 21 columns]

In [152]:
dfiv.tail()

Building ID  Registration ID Borough House Number  \
1657       968093           227740   BRONX          164   
1658       876582           223120   BRONX          201   
1659      1007815           228614   BRONX          157   
1660       968094           227738   BRONX          166   
1661       876582           223120   BRONX          201   

                Street Name  Postcode  # of Dwelling Units  \
1657  WEST KINGSBRIDGE ROAD     10463                   10   
1658  WEST KINGSBRIDGE ROAD     10463                   55   
1659  WEST KINGSBRIDGE ROAD     10463                   20   
1660  WEST KINGSBRIDGE ROAD     10463                   10   
1661  WEST KINGSBRIDGE ROAD     10463                   54   

      Infested Dwelling Unit Count  Eradicated Unit Count  \
1657                             0                      0   
1658                             0                      0   
1659                             0                      0   
1660                             0                      0   
1661                             2                      2   

      Re-infested  Dwelling Unit Count  ... Filing Period Start Date  \
1657                                 0  ...                11/1/2020   
1658                                 0  ...                11/1/2020   
1659                                 0  ...                11/1/2021   
1660                                 0  ...                11/1/2021   
1661                                 0  ...                11/1/2019   

     Filling Period End Date   Latitude  Longitude  Community Board  \
1657              10/31/2021  40.870828 -73.904491                7   
1658              10/31/2021  40.871552 -73.904446                8   
1659              10/31/2022  40.870130 -73.904192                8   
1660              10/31/2022  40.870877 -73.904487                7   
1661              10/31/2020  40.871552 -73.904446                8   

      Council District  2010 Census Tract        BIN           BBL  \
1657                14                269  2128680.0  2.032400e+09   
1658                14                273  2114119.0  2.032560e+09   
1659                14                273  2000000.0  2.032530e+09   
1660                14                269  2128681.0  2.032400e+09   
1661                14                273  2114119.0  2.032560e+09   

                                            NTA  
1657         University Heights (North)-Fordham  
1658  Kingsbridge Heights-Van Cortlandt Village  
1659  Kingsbridge Heights-Van Cortlandt Village  
1660         University Heights (North)-Fordham  
1661  Kingsbridge Heights-Van Cortlandt Village  

[5 rows x 21 columns]